pkg/filesystem/local_directory_darwin.go

```diff
@@ -17,3 +18,7 @@ type deviceNumber = int32
 func (d *localDirectory) Mknod(name path.Component, perm os.FileMode, dev int) error {
 	return status.Error(codes.Unimplemented, "Creation of device nodes is not supported on Darwin")
 }
+
+func clonefileImpl(oldFd int, oldName string, newFd int, newName string) error {
+	return unix.Clonefileat(oldFd, oldName, newFd, newName, 0)
```

Comment:
Please use `CLONE_NOFOLLOW` for the flags. This API isn't supposed to follow symlinks at all, as that would allow you to escape the directory.